h2o-core/src/main/java/water/FailedNodeWatchdogExtension.java

```diff
@@ -227,10 +244,19 @@ public void run() {
                             Log.warn("Watchdog client " + client + " disconnected!");
                             WatchdogClientDisconnectedTask tsk = new WatchdogClientDisconnectedTask(client, watchdogClientRetryTimeout);
                             Log.warn("Asking the rest of the nodes in the cloud whether watchdog client is really gone.");
-                            if((tsk.doAllNodes()).clientDisconnectedConsensus) {
-                                Log.fatal("Stopping H2O cloud since the watchdog client is disconnected from all nodes in the cluster!");
-                                // we should fail with negative status as this is not planned shutdown
-                                H2O.shutdown(-1);
+                            List<H2ONode> nodesWithClient = tsk.doAllNodes().getNodesWithConnectedClient();
+                            if(nodesWithClient.isEmpty()) {
+                                // we let the leader do the work to have a single point of shutdown initialization
+                                // Note: we know leader is working properly because it responded to MRTask
+                                if (H2O.SELF == H2O.CLOUD.leader()) {
```

Comment:
Maybe in this case it's irrelevant, but is a good practice to compare `H2ONode`s based on references?

Reply:
This is safe to do because all the nodes are interned. And you can be sure that SELF is in `H2O.CLOUD._memary`.

Reply:
Ok, make sense. Thanks!